ads/map_impl.go

```diff
@@ -257,3 +269,17 @@ func (m *authenticatedMap[IdentifierType, K, V]) has(keyBytes []byte) (has bool,
 
 	return value != nil, nil
 }
+
+func (m *authenticatedMap[IdentifierType, K, V]) addSize(delta int) error {
+
+	size, err := m.size.Get()
+	if err != nil && !ierrors.Is(err, kvstore.ErrKeyNotFound) {
```

Comment:
should we maybe fall back to 0 here? You do it in `Size()` too.

Reply:
Here we have the context to handle errors properly though. We fall back to 0 if the key is not found. But if there's any other error doesn't that mean there's something wrong with the KV store? The subsequent write would fail as well.